numba/tests/test_array_methods.py

```diff
@@ -491,6 +511,12 @@ def check(arr, dtype):
             check(arr, dt)
         self.assertIn('cannot convert from int32 to Record',
                       str(raises.exception))
+        # Non-Literal String
+        unicode_val = "float32"
+        with self.assertTypingError() as raises:
+            check_dtype_arg(arr, unicode_val)
+        self.assertIn('array.astype if dtype is a string it must be constant',
+                      str(raises.exception))
```

Comment:
Suggest something like this as it's really only this one case that exists?:
```diff
diff --git a/numba/tests/test_array_methods.py b/numba/tests/test_array_methods.py
index 64c0eba..c130d88 100644
--- a/numba/tests/test_array_methods.py
+++ b/numba/tests/test_array_methods.py
@@ -469,25 +469,6 @@ class TestArrayMethods(MemoryLeakMixin, TestCase):
             got = run(arr, dtype)
             self.assertPreciseEqual(got, expected)
 
-        def run_dtype_arg(arr, dtype):
-            """
-            Equivalent to run but dtype is passed as
-            an arg instead of as a lowered constant.
-            This is used to check LiteralValue requirements.
-            """
-            pyfunc = lambda arr, dtype: arr.astype(dtype)
-            cres = self.ccache.compile(pyfunc, (typeof(arr), typeof(dtype)))
-            return cres.entry_point(arr, dtype)
-        def check_dtype_arg(arr, dtype):
-            """
-            Equivalent to check but dtype is passed as
-            an arg instead of as a lowered constant.
-            This is used to check LiteralValue requirements.
-            """
-            expected = arr.astype(dtype).copy(order='A')
-            got = run_dtype_arg(arr, dtype)
-            self.assertPreciseEqual(got, expected)
-
         # C-contiguous
         arr = np.arange(24, dtype=np.int8)
         check(arr, np.dtype('int16'))
@@ -515,10 +496,14 @@ class TestArrayMethods(MemoryLeakMixin, TestCase):
             check(arr, dt)
         self.assertIn('cannot convert from int32 to Record',
                       str(raises.exception))
-        # Non-Literal String
+
+        # Check non-Literal string raises
         unicode_val = "float32"
         with self.assertTypingError() as raises:
-            check_dtype_arg(arr, unicode_val)
+            @jit(nopython=True)
+            def foo(dtype):
+                np.array([1]).astype(dtype)
+            foo(unicode_val)
         self.assertIn('array.astype if dtype is a string it must be constant',
                       str(raises.exception))
```